src/modules/vtol_att_control/tiltrotor.cpp

```diff
@@ -266,35 +267,91 @@ void Tiltrotor::update_external_state()
 
 }
 
- /**
-* Prepare message to acutators with data from the attitude controllers.
+/**
+* Write data to actuator output topic.
 */
-void Tiltrotor::fill_att_control_output()
+void Tiltrotor::fill_actuator_outputs()
 {
-	_actuators_out_0->control[0] = _actuators_mc_in->control[0] * _roll_weight_mc;	// roll
-	_actuators_out_0->control[1] = _actuators_mc_in->control[1] * _roll_weight_mc;	// pitch
-	_actuators_out_0->control[2] = _actuators_mc_in->control[2] * _roll_weight_mc;	// yaw
+	switch(_vtol_schedule.flight_mode) {
+		case MC_MODE:
+			_actuators_out_0->control[0] = _actuators_mc_in->control[0];
+			_actuators_out_0->control[1] = _actuators_mc_in->control[1];
+			_actuators_out_0->control[2] = _actuators_mc_in->control[2];
+			_actuators_out_0->control[3] = _actuators_mc_in->control[3];
+			_actuators_out_1->control[0] = 0;
+			_actuators_out_1->control[1] = 0;
+			_actuators_out_1->control[4] = _tilt_control;
+			break;
+		case FW_MODE:
+			_actuators_out_0->control[0] = 0;
+			_actuators_out_0->control[1] = 0;
+			_actuators_out_0->control[2] = 0;
+			_actuators_out_0->control[3] = _actuators_fw_in->control[3];
+
+			_actuators_out_1->control[0] = -_actuators_fw_in->control[0];	// roll elevon
+			_actuators_out_1->control[1] = _actuators_fw_in->control[1] + _params->fw_pitch_trim;	// pitch elevon
+			_actuators_out_1->control[2] = _actuators_fw_in->control[2];	// yaw
+			_actuators_out_1->control[3] = _actuators_fw_in->control[3];	// throttle
+			_actuators_out_1->control[4] = _tilt_control;
+			break;
+		case TRANSITION_FRONT_P1:
+			_actuators_out_0->control[0] = _actuators_mc_in->control[0] * _roll_weight_mc;
+			_actuators_out_0->control[1] = _actuators_mc_in->control[1];
+			_actuators_out_0->control[2] = _actuators_mc_in->control[2] * _yaw_weight_mc;
+			_actuators_out_0->control[3] = _actuators_mc_in->control[3];
+			_actuators_out_1->control[0] = -_actuators_fw_in->control[0] * (1.0f - _roll_weight_mc);	//roll elevon
+			_actuators_out_1->control[1] = (_actuators_fw_in->control[1] + _params->fw_pitch_trim);		//pitch elevon
+			_actuators_out_1->control[4] = _tilt_control;	// for tilt-rotor control
+			break;
+		case TRANSITION_FRONT_P2:
```

Comment:
I agree, I initially wasn't sure that it would be like that. Let's do the test and if this works, we can very much reduce the function.